utils/autotailor

```diff
@@ -71,45 +156,58 @@ class Tailoring:
     def add_value_change(self, varname, value):
         self.value_changes.append((varname, value))
 
+    def _add_rule_select_operations(self, container_element):
+        for rule_id in self.rules_to_select:
+            change = ET.SubElement(container_element, "{%s}select" % NS)
+            change.set("idref", self._full_rule_id(rule_id))
+            change.set("selected", "true")
+
+        for rule_id in self.rules_to_unselect:
+            change = ET.SubElement(container_element, "{%s}select" % NS)
+            change.set("idref", self._full_rule_id(rule_id))
+            change.set("selected", "false")
+
+    def _add_value_selections(self, container_element):
+        for varname, value in self.value_changes:
+            change = ET.SubElement(container_element, "{%s}set-value" % NS)
+            change.set("idref", self._full_var_id(varname))
+            change.text = str(value)
+
+    def rule_refinements_to_xml(self, profile_el):
+        for rule_id, refinements in self._rule_refinements.items():
+            ref_rule_el = ET.SubElement(profile_el, "{%s}refine-rule" % NS)
+            ref_rule_el.set("idref", rule_id)
+            for attr, val in refinements.items():
+                ref_rule_el.set(attr, val)
+
     def to_xml(self, location=None):
-        root = ET.Element("xccdf-1.2:Tailoring")
-        root.set("xmlns:xccdf-1.2", ALL_NS["xccdf-1.2"])
-        root.set("id", "xccdf_" + self.id_suffix)
+        root = ET.Element("{%s}Tailoring" % NS)
+        root.set("id", self.id)
 
-        benchmark = ET.SubElement(root, "xccdf-1.2:benchmark")
-        datastream_uri = pathlib.Path(self.original_ds_filename).absolute().as_uri()
+        benchmark = ET.SubElement(root, "{%s}benchmark" % NS)
+        datastream_uri = pathlib.Path(
+            self.original_ds_filename).absolute().as_uri()
         benchmark.set("href", datastream_uri)
 
-        version = ET.SubElement(root, "xccdf-1.2:version")
+        version = ET.SubElement(root, "{%s}version" % NS)
```

Comment:
And this:
```suggestion
        version = ET.SubElement(root, f"{NS}version")
```

Reply:
You actually need to escape the braces: `f"{{{NS}}}version"`, so here I prefer to use the `%`  form.

Reply:
Right.